rwgame/GameWindow.cpp

```diff
@@ -0,0 +1,101 @@
+#include <png.h>
+#include <fstream>
+
+#include <core/Logger.hpp>
+#include "GameWindow.hpp"
+
+GameWindow::GameWindow() :
+	window(nullptr), glcontext(nullptr)
+{
+
+}
+
+
+void GameWindow::create(size_t w, size_t h, bool fullscreen)
+{
+	uint32_t style = SDL_WINDOW_OPENGL/* | SDL_WINDOW_ALLOW_HIGHDPI*/;
+	if (fullscreen)
+		style |= SDL_WINDOW_FULLSCREEN;
```

Comment:
The default style should include `SDL_WINDOW_RESIZABLE`, since that is very useful for debugging (and there's no technical reason why we can't resize the window).

`SDL_WINDOW_FULLSCREEN_DESKTOP` would be nice to use also, but that will mean the window is always at desktop resolution, we'll need some internal framebuffer scaling to match the requested resolution, so this can be done later.